biochemist-python/chapters/Processing_Multiple_Files_and_Writing_Files.ipynb

```diff
@@ -0,0 +1,427 @@
+    "\n",
+    "To make the printing neater, we will separate the PDB ID from the resolution using a tab. To insert a tab, we use the special character `\\t`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "datafile = open('resolutions.txt', 'w+')  #This opens the file for writing\n",
+    "\n",
+    "for f in filenames:\n",
+    "    # Get the PDB ID\n",
+    "    file_name = os.path.basename(f)\n",
+    "    split_filename = file_name.split('.')\n",
+    "    molecule_name = split_filename[0]\n",
+    "    \n",
+    "    # Read the data\n",
+    "    outfile = open(f,'r')\n",
+    "    data = outfile.readlines()\n",
+    "    outfile.close()    \n",
+    "    \n",
+    "    # Loop through the data\n",
+    "    for line in data:\n",
+    "        if 'RESOLUTION.' in line:\n",
+    "            res_line = line\n",
+    "            words = res_line.split()\n",
+    "            resolution = float(words[3])\n",
+    "            datafile.write(F'{molecule_name} \\t {resolution} \\n')\n",
+    "            \n",
+    "datafile.close()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After you run this command, look in the directory where you ran your code and find the \"resolutions.txt\" file.  Open it in a text editor and look at the file.\n",
+    "\n",
+    "In the file writing line, notice the `\\n` at the end of the line.  This is the newline character.  Without it, the text in our file would just be all smushed together on one line.  Also, the `filehandle.close()` command is very important.  Think about a computer as someone who has a very good memory, but is very slow at writing.  Therefore, when you tell the computer to write a line, it remembers what you want it to write, but it doesn't actually write the new file until you tell it you are finished.   The `datafile.close()` command tells the computer you are finished giving it lines to write and that it should go ahead and write the file now.  If you are trying to write a file and the file keeps coming up empty, it is probably because you forgot to close the file.  "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## A final note about string formatting\n",
+    "The F'string' notation that you can use with the print or the write command lets you format strings in many ways.  You could include other words or whole sentences.  For example, we could change the file writing line to\n",
+    "\n",
+    "`datafile.write(F'For the PDB ID {molecule_name} the resolution is {resolution} in Angstroms.')`\n",
+    "\n",
+    "where anything in the braces is a python variable and it will print the value of that variable. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<div class=\"exercise-this-is-a-title exercise\">\n",
+    "<p class=\"exercise-title\">Exercise</p>\n",
```

Comment:
Consider calling this something other than "exercise" if you think it will take a while.

Reply:
I changed _Exercise_ to _Project_.